File: app/src/main/java/com/project/scratchstudio/kith_andoid/Activities/ChangePasswordActivity.java
package com.project.scratchstudio.kith_andoid.Activities;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.net.ConnectivityManager;
import android.os.SystemClock;
import android.preference.PreferenceManager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.KeyEvent;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import com.project.scratchstudio.kith_andoid.CustomViews.CustomFontEditText;
import com.project.scratchstudio.kith_andoid.CustomViews.CustomFontTextView;
import com.project.scratchstudio.kith_andoid.R;
import com.project.scratchstudio.kith_andoid.Service.HttpService;

import java.util.Objects;

public class ChangePasswordActivity extends AppCompatActivity {

    private boolean isChange = false;
    private static final int PASSWORD_CHANGED = 0;
    private static final int PASSWORD_NOT_CHANGED = 1;
    private static long buttonCount = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_change_password);

    }

    public void onClickChangePassword(View view) {
        if (SystemClock.elapsedRealtime() - buttonCount < 1000){
            return;
        }
        buttonCount = SystemClock.elapsedRealtime();
        view.setEnabled(false);
        CustomFontEditText oldPassword = findViewById(R.id.old_password);
        if(Objects.requireNonNull(oldPassword.getText()).toString().equals(HomeActivity.getMainUser().getPassword())){
            CustomFontEditText newPassword = findViewById(R.id.new_password);
            CustomFontEditText repPassword = findViewById(R.id.rep_password);
            if(Objects.requireNonNull(newPassword.getText()).toString().equals(Objects.requireNonNull(repPassword.getText()).toString())){
                if(isNetworkConnected()){
                    HttpService httpService = new HttpService();
                    httpService.changePassword(this, HomeActivity.getMainUser(), newPassword.getText().toString());
                } else{
                    view.setEnabled(true);
                    Toast.makeText(this, "Нет подключения к интернету", Toast.LENGTH_SHORT).show();
                }
            } else {
                view.setEnabled(true);
                Toast.makeText(this, "Новый пароль введен неверно", Toast.LENGTH_SHORT).show();
            }
        } else {
            view.setEnabled(true);
            Toast.makeText(this, "Старый пароль введен неверно", Toast.LENGTH_SHORT).show();
        }
    }

    public void saveNewPassword(String password){
        Button button = findViewById(R.id.change);
        button.setEnabled(true);
        SharedPreferences sp = PreferenceManager.getDefaultSharedPreferences(this);
        SharedPreferences.Editor ed;
        ed = sp.edit();
        ed.putString("cur_user_password", password);
        ed.apply();

        HomeActivity.getMainUser().setPassword(password);
        isChange = true;
    }

    @Override
    public boolean onKeyDown(int keyCode, KeyEvent event)  {
        if (keyCode == KeyEvent.KEYCODE_BACK ) {
            Intent intent = new Intent(ChangePasswordActivity.this, EditActivity.class);
            if(isChange)
                setResult(PASSWORD_CHANGED, intent);
            else
                setResult(PASSWORD_NOT_CHANGED, intent);
            finish();
            return true;
        }

        return super.onKeyDown(keyCode, event);
    }

    private boolean isNetworkConnected() {
        ConnectivityManager cm = (ConnectivityManager) getSystemService(Context.CONNECTIVITY_SERVICE);
        return cm.getActiveNetworkInfo() != null;
    }

    public void onClickCancel(View view) {
        if (SystemClock.elapsedRealtime() - buttonCount < 1000) return;
        buttonCount = SystemClock.elapsedRealtime();
        Intent intent = new Intent(ChangePasswordActivity.this, EditActivity.class);
        if(isChange)
            setResult(PASSWORD_CHANGED, intent);
        else
            setResult(PASSWORD_NOT_CHANGED, intent);
        finish();
    }

    public void onClickDone(View view){
        if (SystemClock.elapsedRealtime() - buttonCount < 1000) return;
        buttonCount = SystemClock.elapsedRealtime();
        Intent intent = new Intent(ChangePasswordActivity.this, EditActivity.class);
        if(isChange)
            setResult(PASSWORD_CHANGED, intent);
        else
            setResult(PASSWORD_NOT_CHANGED, intent);
        finish();
    }

}

File: app/src/main/java/com/project/scratchstudio/kith_andoid/activities/ChangePasswordActivity.kt
package com.project.scratchstudio.kith_andoid.activities

import android.accounts.NetworkErrorException
import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import android.net.ConnectivityManager
import android.os.Bundle
import android.os.SystemClock
import android.preference.PreferenceManager
import android.text.Editable
import android.view.KeyEvent
import android.view.View
import android.widget.Button
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.project.scratchstudio.kith_andoid.custom_views.to_trash.CustomFontEditText
import com.project.scratchstudio.kith_andoid.R
import com.project.scratchstudio.kith_andoid.UserPresenter
import com.project.scratchstudio.kith_andoid.network.model.BaseResponse
import java.util.Objects

class ChangePasswordActivity : AppCompatActivity() {

    private var isChange = false
    private lateinit var userPresenter: UserPresenter

    private val isNetworkConnected: Boolean
        get() {
            val cm = getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            return cm.activeNetworkInfo != null
        }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_change_password)
        userPresenter = UserPresenter(applicationContext)
    }

    fun onClickChangePassword(view: View) {
        if (SystemClock.elapsedRealtime() - buttonCount < 1000) {
            return
        }
        buttonCount = SystemClock.elapsedRealtime()
        view.isEnabled = false
        val oldPassword = findViewById<CustomFontEditText>(R.id.old_password)
        if (Objects.requireNonNull<Editable>(oldPassword.text).toString() == HomeActivity.mainUser.password) {
            val newPassword = findViewById<CustomFontEditText>(R.id.new_password)
            val repPassword = findViewById<CustomFontEditText>(R.id.rep_password)
            if (Objects.requireNonNull<Editable>(newPassword.text).toString() == Objects.requireNonNull<Editable>(repPassword.text).toString()) {
                if (isNetworkConnected) {
                    userPresenter.changePassword(object : UserPresenter.BaseCallback {
                        override fun onSuccess(baseResponse: BaseResponse) {
                            if (baseResponse.status) {
                                Toast.makeText(applicationContext, "Пароль изменен", Toast.LENGTH_SHORT).show()
                                saveNewPassword(newPassword.text.toString())
                            } else {
                                Toast.makeText(applicationContext, "Ошибка отправки запроса", Toast.LENGTH_SHORT).show()
                            }
                            view.isEnabled = true
                        }

                        override fun onError(networkError: NetworkErrorException) {
                            Toast.makeText(applicationContext, "Ошибка отправки запроса", Toast.LENGTH_SHORT).show()
                            view.isEnabled = true
                        }
                    }, HomeActivity.mainUser.id, HomeActivity.mainUser.password, newPassword.text.toString())
                } else {
                    view.isEnabled = true
                    Toast.makeText(this, "Нет подключения к интернету", Toast.LENGTH_SHORT).show()
                }
            } else {
                view.isEnabled = true
                Toast.makeText(this, "Новый пароль введен неверно", Toast.LENGTH_SHORT).show()
            }
        } else {
            view.isEnabled = true
            Toast.makeText(this, "Старый пароль введен неверно", Toast.LENGTH_SHORT).show()
        }
    }

    fun saveNewPassword(password: String) {
        val button = findViewById<Button>(R.id.change)
        button.isEnabled = true
        val sp = PreferenceManager.getDefaultSharedPreferences(this)
        val ed: SharedPreferences.Editor
        ed = sp.edit()
        ed.putString("cur_user_password", password)
        ed.apply()

        HomeActivity.mainUser.password = password
        isChange = true
    }

    override fun onKeyDown(keyCode: Int, event: KeyEvent): Boolean {
        if (keyCode == KeyEvent.KEYCODE_BACK) {
            val intent = Intent(this@ChangePasswordActivity, EditActivity::class.java)
            if (isChange)
                setResult(PASSWORD_CHANGED, intent)
            else
                setResult(PASSWORD_NOT_CHANGED, intent)
            finish()
            return true
        }

        return super.onKeyDown(keyCode, event)
    }

    fun onClickCancel() {
        if (SystemClock.elapsedRealtime() - buttonCount < 1000) return
        buttonCount = SystemClock.elapsedRealtime()
        val intent = Intent(this@ChangePasswordActivity, EditActivity::class.java)
        if (isChange)
            setResult(PASSWORD_CHANGED, intent)
        else
            setResult(PASSWORD_NOT_CHANGED, intent)
        finish()
    }

    fun onClickDone() {
        if (SystemClock.elapsedRealtime() - buttonCount < 1000) return
        buttonCount = SystemClock.elapsedRealtime()
        val intent = Intent(this@ChangePasswordActivity, EditActivity::class.java)
        if (isChange)
            setResult(PASSWORD_CHANGED, intent)
        else
            setResult(PASSWORD_NOT_CHANGED, intent)
        finish()
    }

    companion object {
        private val PASSWORD_CHANGED = 0
        private val PASSWORD_NOT_CHANGED = 1
        private var buttonCount: Long = 0
    }

}
